src/Microsoft.Azure.ServiceBus/MessageSender.cs

```diff
@@ -35,18 +35,55 @@ public Task SendAsync(IEnumerable<BrokeredMessage> brokeredMessages)
             return this.OnSendAsync(brokeredMessages);
         }
 
+        public Task<long> ScheduleMessageAsync(BrokeredMessage message, DateTimeOffset scheduleEnqueueTimeUtc)
+        {
+            if (message == null)
+            {
+                throw Fx.Exception.ArgumentNull(nameof(message));
+            }
+
+            if (scheduleEnqueueTimeUtc.CompareTo(DateTimeOffset.UtcNow) < 0)
+            {
+                throw Fx.Exception.ArgumentOutOfRange(
+                    nameof(scheduleEnqueueTimeUtc),
+                    scheduleEnqueueTimeUtc.ToString(),
+                    "Cannot schedule messages in the past");
+            }
+
+            message.ScheduledEnqueueTimeUtc = scheduleEnqueueTimeUtc.UtcDateTime;
+            MessageSender.ValidateMessage(message);
```

Comment:
Updated it to 'Cannot send a message that was already received.'
Keeping it generic wrt send / schedule.